substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java

```diff
@@ -321,6 +323,9 @@ static boolean isTrustedCryptoProvider(Provider provider) {
 @SuppressWarnings({"unused"})
 final class Target_javax_crypto_JceSecurity {
 
+    @Alias @TargetElement(onlyWith = JDK17_0_10OrLater.class)//
+    public static ReferenceQueue<Object> queue;
```

Comment:
Instead of fixing on the update version, could we create a boolean supplier class, which tests for the `queue` static field in `JceSecurity` class instead?

I.e. change this to:

```
@Alias @TargetElement(onlyWith = JceSecurityHasRefQueueField.class)
public static ReferenceQueue<Object> queue;
```

Thoughts?

Reply:
Yes that makes more sense, we can do the same for the `(Weak)IdentityWrapper` classes. On it.

Reply:
Done